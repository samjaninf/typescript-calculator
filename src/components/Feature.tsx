import { StyleAttribute, css, media } from 'glamor'
import { h } from 'preact'
import { small } from '../styles/breakpoints'
import Checkmark from './Checkmark'
import Soon from './Soon'

export enum Emoji { Checkmark, Soon }

type FeatureProps = {
  emoji: Emoji,
  text: string,
}

const feature: StyleAttribute = css(
  {
    alignItems: 'center',
    display: 'flex',
    flexBasis: '40%',
    height: '2em',
  },
  media(small, { flexBasis: '35%' }),
)

const Feature = ({ emoji, text }: FeatureProps): JSX.Element => (
  <div className={feature.toString()}>
    {emoji === Emoji.Soon ? <Soon /> : <Checkmark />}
    <h3>{text}</h3>
  </div>
)

export default Feature
