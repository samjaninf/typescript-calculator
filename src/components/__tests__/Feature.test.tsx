import Feature, { Emoji } from '../Feature'
import snapshot from './utils/snapshot'

describe('Feature', () => {
  it('renders', () => snapshot(Feature))
  it('renders text property', () => snapshot(Feature, { text: 'Awesome' }))
  it('renders with Checkmark emoji', () => snapshot(Feature, { emoji: Emoji.Checkmark }))
  it('renders with Soon emoji', () => snapshot(Feature, { emoji: Emoji.Soon }))
})
