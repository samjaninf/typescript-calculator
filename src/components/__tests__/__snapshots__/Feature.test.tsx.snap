// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature renders 1`] = `
"<div class=\\"css-g7vt1j\\">
  <Checkmark></Checkmark>
  <h3></h3>
</div>"
`;

exports[`Feature renders text property 1`] = `
"<div class=\\"css-g7vt1j\\">
  <Checkmark></Checkmark>
  <h3>Awesome</h3>
</div>"
`;

exports[`Feature renders with Checkmark emoji 1`] = `
"<div class=\\"css-g7vt1j\\">
  <Checkmark></Checkmark>
  <h3></h3>
</div>"
`;

exports[`Feature renders with Soon emoji 1`] = `
"<div class=\\"css-g7vt1j\\">
  <Soon></Soon>
  <h3></h3>
</div>"
`;
